packages/dds/tree/src/shared-tree/sharedTree-eventing.md

```diff
@@ -0,0 +1,400 @@
+
+Provide a summary of the counts of inserted/removed/moved nodes. Although this will give us the least information out of the three solutions, it may have use cases without needing much additional resources.
+
+```ts
+interface SubtreeAggregateSummaryV1 {
+	version: 1
+	root: UpPath;
+	counts: {
+		inserted: number;
+		deleted: number;
+		moved: number;
+		set: number;
+	};
+}
+```
+
+---
+
+##### Configuration Options
+- maxNodesInDiff/maxDepth – if we want to bound it for large subtrees.
+- includeDiffPerNode – if we want the actual changes, or just if they changed
+- getSubtreeDiff() – to only compute the diffs when necessary
+- getSubtreeJsonPatch() – if we want a standardized format for the diffs
+
+```ts
+type TreeChangedEvent = {
+	node: NodeHandle;
+	getSubtreeDiff?: () => SparseSubtreeV1 | FlatSubtreeV1 | SubtreeAggregate | undefined;
+	getSubtreeJsonPatch?: () => JsonPatchOp[] | undefined
+}
+```
+
+#### Tradeoffs
+| Approach        | Size  | Detail  | Ideal for                                  |
+|-----------------|-------|---------|---------------------------------------------|
+| Sparse Subtree  | Large | High    | Edit information reliant use cases          |
+| Flat List       | Medium| Medium  | Checking which nodes changed                |
+| Aggregate       | Small | Low     | metrics, dashboards                         |
+
+---
+
+## 4. Detached Tree Eventing
+
+### Problem
+
+Detached trees are currently not able to emit events. When editing for detached trees are enabled, we should ensure that eventing for detached trees reaches parity with our current live-tree eventing system. Note that detached trees do not have “root replacement” semantics. EventParent is only used to determine the event source, not to support root replacement logic for detached trees.
+
+### Potential Solution
+
+#### EventParent
+To unify handling of events for nodes that do not have a true parent (root node, detached nodes), we may also introduce an internal helper abstraction called EventParent. 
+
+```ts
+interface EventParent{
+	branch:TreeView | TreeBranch
+	node?: TreeNode
+	rootFieldKey?: FieldKey
+	detachedRootId?: DetachedNodeId
+}
+```
+
+This API would return some dummy parent object (EventParent) that has all the necessary components to track changes on its child node. If the node is parented under another node, we can simply use its parent node. 
+
+However, for nodes without a parent node (detached trees / root node), one potential solution could be to design some sort of object that contains different properties for the different scenarios. 
+
+For instance, although root nodes and detached nodes do not have a parent node, it is still technically parented by the view/branch. For these cases, we could have the object contain the view/branch (and maybe the detachedRootId/rootFieldKey) to identify what exactly we are trying to subscribe to. We can return a parent-like object for eventing purposes, to cover all different scenarios.
```

Comment:
What _are_ the "different scenarios"? I don't really understand what this API is for or how it is useful. Can you provide motivating example scenarios? Let's see the API in action to demonstrate the why and the how.